exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/__init__.py

```diff
@@ -0,0 +1,208 @@
+
+
+.. _Prometheus: https://prometheus.io/
+.. _OpenTelemetry: https://github.com/open-telemetry/opentelemetry-python/
+
+.. code:: python
+
+    from prometheus_client import start_http_server
+
+    from opentelemetry._metrics import get_meter_provider, set_meter_provider
+    from opentelemetry.exporter.prometheus import PrometheusMetricExporter
+    from opentelemetry.sdk._metrics import MeterProvider
+    from opentelemetry.sdk._metrics.export import PullMetricExporterReader
+
+    # Start Prometheus client
+    start_http_server(port=8000, addr="localhost")
+
+    # exporter to export metrics to Prometheus
+    prefix = "MyAppPrefix"
+    exporter = PrometheusMetricExporter(prefix)
+    reader = PullMetricExporterReader(exporter)
+
+    # Meter is responsible for creating and recording metrics
+    set_meter_provider(MeterProvider(metric_readers=[reader]))
+    meter = get_meter_provider().get_meter(__name__)
+
+    counter = meter.create_counter(
+        "requests",
+        "requests",
+        "number of requests",
+    )
+
+    # Labels are used to identify key-values that are associated with a specific
+    # metric that you want to record. These are useful for pre-aggregation and can
+    # be used to store custom dimensions pertaining to a metric
+    labels = {"environment": "staging"}
+
+    counter.add(25, labels)
+    input("Press any key to exit...")
+
+API
+---
+"""
+
+import collections
+import logging
+import re
+from typing import Optional, Sequence, Tuple
+
+from prometheus_client import core
+
+from opentelemetry.sdk._metrics.export import (
+    MetricExportResult,
+    PullMetricExporter,
+)
+from opentelemetry.sdk._metrics.point import Gauge, Histogram, Metric, Sum
+
+_logger = logging.getLogger(__name__)
+
+
+def _convert_buckets(metric: Metric) -> Sequence[Tuple[str, int]]:
+    buckets = []
+    total_count = 0
+    for index, value in enumerate(metric.point.bucket_counts):
+        total_count += value
+        buckets.append(
+            (
+                f"{metric.point.explicit_bounds[index]}",
+                total_count,
+            )
+        )
+    return buckets
+
+
+class PrometheusMetricExporter(PullMetricExporter):
+    """Prometheus metric exporter for OpenTelemetry.
+
+    Args:
+        prefix: single-word application prefix relevant to the domain
+            the metric belongs to.
+    """
+
+    def __init__(self, prefix: str = "") -> None:
+        super().__init__()
+        self._collector = _CustomCollector(self.collect, prefix)
+        core.REGISTRY.register(self._collector)
+
+    def export(self, metrics: Sequence[Metric]) -> MetricExportResult:
+        self._collector.add_metrics_data(metrics)
+        return MetricExportResult.SUCCESS
+
+    def shutdown(self) -> None:
+        core.REGISTRY.unregister(self._collector)
+
+
+class _CustomCollector:
+    """_CustomCollector represents the Prometheus Collector object
+    https://github.com/prometheus/client_python#custom-collectors
```

Comment:
I'm a fan of this way of formatting comments, but this is only a nit:

```suggestion
    """_CustomCollector represents the Prometheus Collector object
    
    See more:
    https://github.com/prometheus/client_python#custom-collectors
```